ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -406,20 +415,64 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
                     )
 
             # L2 Vlans per Tenant
-            # If multiple L2 Vlans share the same name, they will be part of the same bundle
-            for bundle_name, l2vlans in groupby(tenant["l2vlans"], "name"):
+            # If multiple L2 Vlans share the same evpn_vlan_bundle name, they will be part of the same vlan-aware-bundle else they use the vlan name as bundle
+            sorted_vlan_list = sorted(tenant["l2vlans"], key=self._get_vlan_aware_bundle_name_tuple)
+            bundle_groups = itertools_groupby(sorted_vlan_list, self._get_vlan_aware_bundle_name_tuple)
+            for vlan_aware_bundle_name_tuple, l2vlans in bundle_groups:
+                bundle_name, is_evpn_vlan_bundle = vlan_aware_bundle_name_tuple
                 l2vlans = list(l2vlans)
-                if (bundle := self._router_bgp_vlans_vlan(l2vlans[0], tenant, vrf={})) is not None:
+                if is_evpn_vlan_bundle:
+                    # If "evpn_vlan_bundle" in l2vlan bundle group, we use the same logic for rd/rt as VRFs
+                    # using the settings under the given evpn_vlan_bundle only.
+
+                    # check if the referred name exists in the global evpn_vlan_bundles
+                    if (evpn_vlan_bundle := get_item(self._hostvars["evpn_vlan_bundles"], "name", bundle_name)) is None:
+                        raise AristaAvdMissingVariableError(
+                            "The 'evpn_vlan_bundle' of the l2vlans must be defined in the common 'evpn_vlan_bundles' setting. First occurence seen for l2vlan"
+                            f" {l2vlans[0]['id']} in Tenant '{l2vlans[0]['tenant']}' and evpn_vlan_bundle '{l2vlans[0]['evpn_vlan_bundle'].replace('_','')}'."
+                        )
```

Comment:
No need for the replace anymore after the refactor
```suggestion
                            f" {l2vlans[0]['id']} in Tenant '{l2vlans[0]['tenant']}' and evpn_vlan_bundle '{l2vlans[0]['evpn_vlan_bundle']}'."
                        )
```